src/plugins/flow.js

```diff
@@ -1162,6 +1162,7 @@ export default function (instance) {
         let state = this.state.clone();
         try {
           let returnType = this.flowParseTypeAnnotation();
+          if (this.canInsertSemicolon()) this.unexpected();
```

Comment:
I don't think so, because any syntax errors are immediately caught at the end of `parseArrow`: https://github.com/danharper/babylon/blob/774bf486fe2f117393dcf5ee14aa2c050ae40dd7/src/plugins/flow.js#L1169-L1175

(which is also why the raise error happens at `1:1` and not at the actual position)

I'm not 100% sure on _why_ it works like this, though.